flintrock/ec2.py

```diff
@@ -79,21 +82,37 @@ def instances(self):
         else:
             return self.slave_instances
 
+    @property
```

Comment:
My intention in recommending a `@property` was that all the logic would be contained within it. Yes, there is a penalty in that the query to AWS will be made (ignoring any caching that boto3 may do) every time you access the property, but that minor downside is outweighed in my view by the clearer API. The property doesn't get called that often.

So I'd say to either put everything in the property method, or go back to having it just be a plain old property like you did before. It's not a big deal either way.